exams/pearson/download.py

```diff
@@ -233,38 +273,90 @@ def process_eac_file(self, extracted_file):  # pylint: disable=no-self-use
             except ExamAuthorization.DoesNotExist:
-                error_message = (
-                    'Unable to find information for authorization_id: `{authorization_id}` and '
-                    'candidate_id: `{candidate_id}` in our system.'
-                ).format(
-                    authorization_id=result.exam_authorization_id,
-                    candidate_id=result.candidate_id
-                )
-                log.error(error_message)
-                messages.append(error_message)
+                messages.append(format_and_log_error(
+                    'Unable to find a matching ExamAuthorization record:',
+                    client_candidate_id=result.client_candidate_id,
+                    client_authorization_id=result.client_authorization_id,
+                    error=result.message,
+                ))
                 continue
 
             if result.status == VCDC_SUCCESS_STATUS:
                 exam_authorization.status = ExamAuthorization.STATUS_SUCCESS
             else:
                 exam_authorization.status = ExamAuthorization.STATUS_FAILED
-                error_message = (
-                    "Exam authorization failed for user `{username}` "
-                    "with authorization id `{authorization_id}`. {error_message}"
-                ).format(
+                messages.append(format_and_log_error(
+                    'ExamAuthorization sync failed:',
+                    username=exam_authorization.user.username,
+                    client_authorization_id=result.client_authorization_id,
+                    error=result.message,
+                ))
+
+            exam_authorization.save()
+
+        return True, messages
+
+    def process_exam_file(self, extracted_file):
+        """
+        Processes a EXAM file extracted from the zip
+
+        Args:
+            extracted_file (zipfile.ZipExtFile): the extracted file-like object
+
+        Returns:
+            (bool, list(str)): bool is True if file processed successfuly, error messages are returned in the list
+        """
+        log.debug('Found EXAM file: %s', extracted_file)
+        results, invalid_rows = EXAMReader().read(extracted_file)
+        messages = self.get_invalid_row_messages(invalid_rows)
+
+        for result in results:
+            try:
+                exam_authorization = ExamAuthorization.objects.get(id=result.client_authorization_id)
+            except ExamAuthorization.DoesNotExist:
+                messages.append(format_and_log_error(
+                    'Unable to find a matching ExamAuthorization record:',
+                    client_candidate_id=result.client_candidate_id,
+                    client_authorization_id=result.client_authorization_id,
+                ))
+                continue
+
+            if result.grade.lower() not in EXAM_GRADES:
+                messages.append(format_and_log_error(
+                    'Unexpected grade value: grade=`{}`',
```

Comment:
Actually ended up getting rid of this altogether (see my simplified rewrite of `format_and_log_error`).